geom_update.py

```diff
@@ -0,0 +1,159 @@
+            if self.conn != None:
+                self.cursor = self.cur
+                if query!= None:
+                    # catch exception for invalid SQL statement
+
+                    try:
+                        logging.debug('Query sent to Database')
+                        self.cursor.execute(query)
+                        self.conn.commit()
+                        # print(query)
+                        try:
+                            result = self.cursor.fetchall()     
+                            logging.debug('Result fetched from Database')
+                            return result
+                        except:
+                            return self.cursor.statusmessage
+                    except Exception as err:
+                        raise err
+
+                else:
+                    raise ValueError("Query is Null")
+            else:
+                print("Database is not connected")
+        except Exception as err:
+            print("Oops ! You forget to have connection first")
+            raise err
+
+    def close_conn(self):
+        """function for clossing connection to avoid memory leaks"""
+
+        # Check if the connection was successful
+        try:
+            if self.conn != None:
+                if self.cursor:
+                    self.cursor.close()
+                    self.conn.close()
+                    logging.debug("Connection closed")
+        except Exception as err:
+            raise err
+
+class Insight:
+    """This class connects to Insight database and responsible for Values derived from database"""
+
+    def __init__(self, parameters=None):
+        self.database = Database(connection.get_connection_param())
+        self.con, self.cur = self.database.connect()
+        self.params = parameters
+
+    def getMax_osm_element_history_id(self):
+        """Function to extract latest maximum osm element id and minimum osm element id present in Osm_element_history Table"""
+        
+        query = f'''
+                select min(id) as minimum , max(id) as maximum from  public.osm_element_history;
+                '''
+        record=self.database.executequery(query)
+        logging.debug(f"""Maximum Osm element history fetched is {record[0][1]} and minimum is  {record[0][0]}""")
+        return record[0][1],record[0][0]
+
+    def update_geom(self,start,end):
+        """Function that updates geometry column of osm_element_history"""
+        base_query = f"""update
+            osm_element_history as oeh
+                            set
+            geom =
+            case
+                when oeh.type = 'node' then ST_MakePoint(oeh.lon,
+                oeh.lat)
+                when oeh.type = 'way' then public.construct_geometry(oeh.nds,
+                oeh.id,
+                oeh."timestamp")
+                else 
+                    Null
+            end
+        where
+            oeh.action != 'delete'
+            and oeh.type != 'relation'"""
```

Comment:
in this case, whenever we run this geom_update script we will update all osm_element_history table again, while we need to construct the ways geom and fix the nodes geom

we can add this condition to the node checking ..  oeh.geom = ST_MakePoint(oeh.lat, oeh.lon)